### PR TITLE
Library

### PR DESCRIPTION
**Pull request by karthi07**
The code has following implementation
 - Added Class for Books
 - constructor for the class
 - functions to addBook and Show book at homepage
 - added Event listener for buttons and form
 - Implemented LocalStorage to save books data

### CODE REVIEW

**Review by shloch — CHANGES_REQUESTED** [review]
## Hello @karthi07 , @evansinho 

Great project and work from you, works great
bok

### [changes required]

- separate LOGIC from DOM MANIPULATION (you can split code into 2 `js` files and include them into `HTML` [see [requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png)]

### [optional changes]
- Add a screenshot of your project inside your README file

### Waiting for the last review hopefully...great job

File: library.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    this.title = title;
+    this.pages = pages;
+    this.read = read;
+  }
+}
+
+function addBookToLibrary(book) {
+  if (book.title === '' || book.author === '' || book.pages === '') {
+    // eslint-disable-next-line no-throw-literal
+    throw 'error';
```
Comment (maelfosso) [inline library.js]: Instead of throwing an error in such an application, it's preferable to display a message to the user.

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello,

Thank you for applying the suggested requested changes.
You code is nicer now.
However, there are some issues that need to be fix.

1. I got this error when wanting to change the book status.
Screenshot at 2020-07-03 06:53:45

2. There is no message, but an error in the console, when the user try to submit an empty form. Please, read the comment left below.

3. Those 2 are logic 
https://github.com/evansinho/Library/blob/ce3fa8cbda20dea7621781df9818bad2982393a0/dom.js#L39
https://github.com/evansinho/Library/blob/ce3fa8cbda20dea7621781df9818bad2982393a0/dom.js#L44
Move them out of DOM please.

Kindly make the changes and submit a new code review request
Thanks

Mael FOSSO

File: dom.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  document.querySelector('#bookTitle').value = '';
+  document.querySelector('#bookPages').value = '';
+  document.querySelector('#bookRead').checked = false;
+}
+
+document.querySelector('#books-ctn').addEventListener('click', (e) => {
+  e.preventDefault();
+  if (e.target.classList.contains('delete')) {
+    const { id } = e.target.parentElement.dataset;
+    myLibrary.splice(id, 1);
```
Comment (maelfosso) [inline dom.js]: Follow the indication given up there, please.

File: dom.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    render();
+  }
+  if (e.target.classList.contains('read')) {
+    const { id } = e.target.parentElement.dataset;
+    myLibrary[id].read = !myLibrary[id].read;
```
Comment (maelfosso) [inline dom.js]: Follow the indications given up there

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required :red_circle: 

Hello coders,

Thank you for applying the changes. However, as indicated in the last review, there are still those two logic into the DOM. Please, move them into `logic.js` file.

Create two methods in `logic.js`: **delete** and **toggleReadStatus** and call them with the good parameters.

Kindly make the changes and submit a new code review request
Thanks

Mael FOSSO

**Review by Oghenebrume50 — APPROVED** [review]
### APPROVED 

Hi @karthi07 @evansinho  this looks good to me :+1: please merge and move on **This is approved**

Nice project though, I love the UI :slightly_smiling_face: 

**RAPHAEL NORIODE** :mask:

**Pull request merged**